js/components/AccountForm.vue

```diff
@@ -0,0 +1,277 @@
+                        <option value="tls">{{ t('mail', 'STARTTLS') }}</option>
+                    </select>
+                </p>
+                <p class="groupmiddle">
+                    <input type="number"
+                        ref="smtpPort"
+                        name="smtp-port"
+                        :placeholder="t('mail', 'SMTP Port')"
+                        :value="config.smtpPort" />
+                </p>
+                <p class="groupmiddle">
+                    <input type="text"
+                        ref="smtpUser"
+                        name="smtp-user"
+                        :placeholder="t('mail', 'SMTP User')"
+                        :value="config.smtpUser" />
+                </p>
+                <p class="groupbottom">
+                    <input type="password"
+                        ref="smtpPassword"
+                        name="smtp-password"
+                        :placeholder="t('mail', 'SMTP Password')"
+                        :value="config.smtpPassword"
+                        required />
+                </p>
+            </div>
+
+            <input type="submit"
+                ref="submitButton"
+                class="primary"
+                :value="t('mail', 'Connect')"/>
+            </fieldset>
+        </form>
+    </div>
+</template>
+
+<script>
+export default {
+  name: 'AccountForm',
+  props: {
+    settingsPage: Boolean
+  },
+  data() {
+    return {
+      firstToggle: true,
+      config: {
+        accountName: $('#user-displayname').text() || '',
+        emailAddress: $('#user-email').text() || '',
+        password: '',
+        accountName: '',
+        autoDetect: true,
+        imapPort: 993,
+        imapSslMode: 'ssl',
+        smtpPort: 587,
+        smtpSslMode: 'tls'
+      }
+    };
+  },
+  mounted() {
+    if (this.settingsPage) {
+      $(this.$refs.emptyContent).hide();
+      $(this.$refs.submitButton).val(t('mail', 'Save'));
+    }
+
+    if (this.config.autoDetect) {
+      $(this.$refs.mailPassword).show();
+      $(this.$refs.manualInputs).hide();
+    } else {
+      $(this.$refs.mailPassword).hide();
+    }
+
+    $(this.$refs.imapSslMode)
+      .find(`[value="${this.config.imapSslMode}"]`)
+      .attr({ selected: 'selected' });
+    $(this.$refs.smtpSslMode)
+      .find(`[value="${this.config.smtpSslMode}"]`)
+      .attr({ selected: 'selected' });
+  },
+  methods: {
+    toggleManualMode: function() {
+      this.config.autoDetect = !this.config.autoDetect;
+
+      $(this.$refs.manualInputs).slideToggle();
+      this.$refs.imapHost.focus();
+
+      if (!this.config.autoDetect) {
+        if (this.firstToggle) {
+          // Manual mode opened for the first time
+          // -> copy email, password for imap&smtp
+          const email = $(this.$refs.mailAddress).val();
+          const password = $(this.$refs.mailPassword).val();
+
+          $(this.$refs.imapUser).val(email);
```

Comment:
We definitely should use Vue form input bindings for these values: https://vuejs.org/v2/guide/forms.html

Reply:
Should be fixed now @ChristophWurst

Reply:
Thanks!